js/stats.js

```diff
@@ -0,0 +1,107 @@
+require.scopes.stats = ( () => {
+    var utils = require('utils'),
+        statsByParentCompany = {},
+        statsBySite = {},
+        topBlocked = [];
+
+    function updateStatsFromTabData(tab){
+        if(tab && tab.trackers && tab.trackers){
+            Object.keys(tab.trackers).forEach(function(parentCompany){
+                update(parentCompany, tab.url, tab.trackers[parentCompany]);
+            });
+        }
+        // only sync when tab load completes. Should help avoid chrome.storage rate limts
+        syncToStorage();
+    }
+
+    function update(parentCompany, currentSite, tracker){
+        currentSite = utils.extractHostFromURL(currentSite);
+        addToParentCompanyStats(parentCompany, tracker);
+        addToSiteStats(currentSite, tracker);
+        calcTopBlocked(parentCompany);
+    }
+
+    function addToParentCompanyStats(parentCompany, tracker){
+        if(statsByParentCompany[parentCompany]){
+            statsByParentCompany[parentCompany] += tracker.count;
+        }
+        else{
+            statsByParentCompany[parentCompany] = tracker.count;
+        }
+    }
+
+    function addToSiteStats(site, tracker){
```

Comment:
this function needs a solid unit test to show that it functions with many possible example and extreme cases.